### DISCUSSION TITLE
waitForTransaction giving error on zksync

### DISCUSSION THREAD

**Discussion by FurkanSezal** [wevm/viem#1207] [Question] [answered]
Hello, i am getting this error only on zksync testnet. Not on every tx. Getting it like 2/5 times. When i get the error i can see the tx hash on the console. And tx is success. Any ideas ? 

```
TypeError: Cannot mix BigInt and other types, use explicit conversions
    at Object.onBlockNumber (waitForTransactionReceipt.js:100:1)
```

`import { waitForTransaction } from 'wagmi/actions';`

```
    const tx = await marketplaceContract.write.cancelOrder([
      {//  parameters...
        },
      },
      sig,
    ]);

    setText('Awaiting blockchain...');
    console.log(tx);
    const data = await waitForTransaction({
      hash: tx,
    });
    console.log(data);
    return tx;
  } catch (err) {
    console.log('error from blockchain operations');
    throw err;
  }
```

**Comment by tmm** [comment, marked as answer]
Check out the latest Viem version. Issue https://github.com/wagmi-dev/viem/issues/1192 was raised for this and it was fixed by https://github.com/wagmi-dev/viem/commit/fa2d87942f792ee3b87c9ac291c2c82f8c2fc313.
